### PR TITLE
Move files from interfaces to designated directories

### PR DESCRIPTION
**Pull request by nickaleks**
hashable, signable, primitive and model_primitive are moved to 'base' directory, polymorphic_wrapper and visitor_apply_for_all to utils.

### CODE REVIEW

File: shared_model/interfaces/query_responses/transactions_response.hpp [hunk 1]
```diff
@@ -20,12 +20,12 @@
 
 #include <rxcpp/rx-observable.hpp>
```
Comment (l4l) [inline shared_model/interfaces/query_responses/transactions_response.hpp]: Remove this line as well plz, shouldn't be here

File: shared_model/cryptography/crypto_provider/crypto_model_signer.hpp [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 #include "cryptography/keypair.hpp"
 #include "interfaces/common_objects/signature.hpp"
-#include "interfaces/signable.hpp"
+#include "interfaces/base/signable.hpp"
```
Comment (l4l) [inline shared_model/cryptography/crypto_provider/crypto_model_signer.hpp]: Check [that](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) out please, especially:
>Within each section the includes should be ordered alphabetically

File: shared_model/interfaces/transaction.hpp [hunk 3]
```diff
@@ -22,11 +22,11 @@
 #include <vector>
```
Comment (l4l) [inline shared_model/interfaces/transaction.hpp]: `unordered_set` above also unused

File: shared_model/cryptography/crypto_provider/crypto_model_signer.hpp (hunk 2 above)
Comment (nickaleks) [reply shared_model/cryptography/crypto_provider/crypto_model_signer.hpp]: Fixed every file affected by changes

File: shared_model/interfaces/query_responses/transactions_response.hpp (hunk 1 above)
Comment (nickaleks) [reply shared_model/interfaces/query_responses/transactions_response.hpp]: done

File: shared_model/interfaces/transaction.hpp (hunk 3 above)
Comment (nickaleks) [reply shared_model/interfaces/transaction.hpp]: done

**Comment by muratovv** [comment]
Moved to https://github.com/hyperledger/iroha/pull/741.

**Pull request closed without merging**